### DISCUSSION TITLE
Finding random point within specific area

### DISCUSSION THREAD

**Discussion by PacaPop** [recastnavigation/recastnavigation#606] [Q&A]
Hello! 

I am needing to get a random point within a specific area. 

I already have the area defined and I have its ID. 
I wanted to create a dtQueryFilter that would allow me to use findRandomPoint() or findRandomPointAroundCircle() to get points that always fall within that area.

How would I have to do to achieve this?

Thanks!!!!!

**Comment by grahamboree** [comment]
Hi!  The easiest way to do this is probably to uncomment [this line](https://github.com/recastnavigation/recastnavigation/blob/405cc095ab3a2df976a298421974a2af83843baf/Detour/Include/DetourNavMeshQuery.h#L31) in `DetourNavMeshQuery.h`, and create a subclass of `dtQueryFilter`.  In the `passFilter` function override, you can filter triangles based on whether they fall within the area or not.
